### PR TITLE
Additional Milesight sensors

### PR DESCRIPTION
**Pull request by J2thatsme**
Added a lot of Milesight Devices

### CODE REVIEW

File: types/milesight/GS101/default.schema.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  "title": "Default",
+  "properties": {
+    "gasState": {
+      "title": "Gas state",
+      "description": "Gas state",
+      "type": "string",
+      "enum": ["NORMAL", "ABNORMAL"]
+    },
+    "valve": {
+      "title": "Gas valve",
```
Comment (mariusgiger) [inline types/milesight/GS101/default.schema.json:16]: Valve state?

File: types/milesight/GS101/default.schema.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      "description": "Gas valve",
+      "type": "string",
+      "enum": ["CLOSED", "OPENED"]
+    },
+    "relay": {
+      "title": "Relay",
```
Comment (mariusgiger) [inline types/milesight/GS101/default.schema.json:22]: Relay state?

File: types/milesight/GS101/lifecycle.schema.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
+{
+  "$id": "https://akenza.io/milesight/GS101/lifecycle.schema.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "processingType": "uplink_decoder",
+  "topic": "lifecycle",
+  "title": "Lifecycle",
+  "properties": {
+    "lifeRemain": {
+      "title": "Life time remain",
```
Comment (mariusgiger) [inline types/milesight/GS101/lifecycle.schema.json:10]: Remaining lifetime, also rename all other instances ;)

File: types/milesight/WS51X/meta.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "WS51X",
+  "version": "1.0.0",
+  "manufacturer": "Milesight",
+  "url": "https://github.com/Milesight-IoT/SensorDecoders/tree/main/WS_Series/WS51x",
```
[the remark this answers is not in the document]
Comment (J2thatsme) [reply types/milesight/WS51X/meta.json:5]: There is no website for those sensors sadly. I could link to an external shop site.

File: types/milesight/AT101/uplink.js [hunk 5]
```diff
@@ -132,16 +132,16 @@ function consume(event) {
...
-      pastLocations.longitude =
-        readInt32LE(bytes.slice(i + 4, i + 8)) / 1000000;
-      pastLocations.latitude =
-        readInt32LE(bytes.slice(i + 8, i + 12)) / 1000000;
+      const timestamp = new Date(readUInt32LE(bytes.slice(i, i + 4)) * 1000);
+      const longitude = readInt32LE(bytes.slice(i + 4, i + 8)) / 1000000;
+      const latitude = readInt32LE(bytes.slice(i + 8, i + 12)) / 1000000;
       i += 12;
 
-      history.locationHistory = history.locationHistory || [];
```
[the remark this answers is not in the document]
Comment (J2thatsme) [reply types/milesight/AT101/uplink.js:143]: Did not see any active ones

**Review by mariusgiger — DISMISSED** [review]

**Review by mariusgiger — APPROVED** [review]

**Pull request merged**